### PR TITLE
Fix Sphinx warnings

### PR DESCRIPTION
**Pull request by cmarqu**
* Always include sphinxcontrib.spelling
* Add files not directly included to exclude_patterns

### CODE REVIEW

File: documentation/source/conf.py [hunk 1]
```diff
@@ -95,7 +96,11 @@
 
 # List of patterns, relative to source directory, that match files and
 # directories to ignore when looking for source files.
-exclude_patterns = []
+exclude_patterns = [
+    "newsfragments/*.rst",
+    "generated/namespacelist.rst",
+    "generated/namespace/*.rst",
```
Comment (eric-wieser) [inline documentation/source/conf.py]: If you know where these come from, can you add a comment?

File: documentation/source/conf.py [hunk 2]
```diff
@@ -95,7 +96,11 @@
 
 # List of patterns, relative to source directory, that match files and
 # directories to ignore when looking for source files.
-exclude_patterns = []
+exclude_patterns = [
+    "newsfragments/*.rst",
```
Comment (eric-wieser) [inline documentation/source/conf.py]:
```suggestion
    # these are compiled into a single file at build-time, so there's no need to build them separately
    "newsfragments/*.rst",
```

File: documentation/source/conf.py (hunk 1 above)
Comment (eric-wieser) [reply documentation/source/conf.py]: I think this is correct?
```suggestion
    # unused outputs from breathe
    "generated/namespacelist.rst",
    "generated/namespace/*.rst",
```

File: documentation/source/conf.py (hunk 2 above)
Comment (cmarqu) [reply documentation/source/conf.py]: Added.

File: documentation/source/conf.py (hunk 1 above)
Comment (cmarqu) [reply documentation/source/conf.py]: It is, thanks.

**Pull request merged**